snippets/survival_analysis/Coxph_model/Coxph_survival_regression.Rmd

```diff
@@ -8,61 +8,90 @@ author: "Jana Fischer"
+ OSmonth: overall suvival from time of diagnosis [months]  
+ DFSmonth: disease free survival from time of diagnosis [months]  
+ Patientstatus: life/death status   
+  "death by primary disease" = death due to disease  
+  "death" = disease unrelated death  
+  "alive" = alive at given survival time  
+  "alive w metastases" = alive with metasatses at given survival time  
+
+
 # Prepare censoring information
-```{r}
 
-#Censoring for overall survival (1 = event (death due to disease), 0 = censored)
+Survival data is censored when the event has not yet occurred for a patient
+at a given time or the patient is lost to follow-up during the study.
+A more detailed explanation can be found here: 
+http://www.sthda.com/english/wiki/survival-analysis-basics#censoring
+
+```{r}
+# Censoring for overall survival (1 = event (death due to disease), 0 = censored)
 surv_dat$censoringOS = 0
 surv_dat$censoringOS[surv_dat$Patientstatus == 'death by primary disease'] = 1
 
-#Censoring for disease free survival (1 = event (recurrence), 0 = censored)
+# Censoring for disease free survival (1 = event (recurrence), 0 = censored)
 surv_dat$censoringDFS = 0
 surv_dat$censoringDFS[surv_dat$Patientstatus == 'alive w metastases'] = 1
 surv_dat$censoringDFS[surv_dat$OSmonth > surv_dat$DFSmonth] = 1 
 
 ```
 
-# Coxph survival model for grade
-```{r}
-library(survival)
-library(ggplot2)
+# Coxph survival model for grade groups
+
+Covariates passed to the coxph model can be continuous or categorical.
+With categorical covariates each level is considered a distinct group
+and the first level is used as reference.
 
+```{r}
 #Set reference level for grade groups (first level)
 surv_dat$grade = factor(surv_dat$grade,levels = 1:3)
 
-#Calculate coxph model
+#Calculate coxph model. The formula requires a standard survival object
+#built with the Surv function from the "survival" package. In the survival
+#object a plus behind the time indicates censoring.
 res = coxph(Surv(surv_dat$OSmonth, surv_dat$censoringOS) ~ grade , data =  surv_dat)
 summary(res)
+```
 
+The output contains:  
+Pr(>|z|): Wald statistic of whether the coefficient is statistically significantly
+different from 0. Grade 3 is significant.  
+coef: For the regression coefficients a positive sign represents higher risk of death and
+a negative sign lower risk. Grade 3 has a higher risk of death compared to grade 1.    
+exp(coef): The exponent of the coefficient is the hazard ratio. Grade 3 patients have a
+risk that is increased by a factor of 9 compared to grade 1.  
+The global statistical significance of the model can be found at the bottom of the output.  
+
+More detailed information and examples of multivariate cox regressions can be found here:
+http://www.sthda.com/english/wiki/cox-proportional-hazards-model
+
+
+# Plot hazard ratios
+```{r}
 #Extract hazard ratios and confidence intervals from model output
 CI = confint(res)
 dt <- data.table(x = names(res$coefficients),
```

Comment:
I really can recommend the `broom` package that provides a lot of functions to extract 'tindy' data from all kind of statistical models - usually in a format ready to plot with ggplot.

General:
https://cran.r-project.org/web/packages/broom/vignettes/broom.html

Coxph:
https://broom.tidyverse.org/reference/tidy.coxph.html